src/App.js

```diff
@@ -1,26 +1,91 @@
 import React from 'react';
-import logo from './logo.svg';
+import foods from './foods.json';
+import FoodBox from './components/FoodBox';
+import MenuDay from './components/MenuDay';
+import AddFood from './components/AddFood';
+import Searchbar from './components/Searchbar';
 import './App.css';
+import 'bulma/css/bulma.css';
+import ButtonAddForm from './components/ButtonAddForm';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+
+class App extends React.Component{
+
+  constructor(props){
+    super(props)
+    this.state ={
+        foods: foods,
+        filteredFood: foods,
+        condition:false,
+        todayFood:[]
+    }
+}
+
+ // function to add new food
+ handleAddFood=(event)=>{
+   event.preventDefault()
+   let newFood = event.currentTarget.food.value
+   let newCalories = event.currentTarget.calories.value
+   let newImage = event.currentTarget.urlImg.value
+   let newFoods =[{name:newFood,calories:newCalories,image:newImage},...this.state.foods]
+   this.setState({
+     foods:newFoods,
+     filteredFood:newFoods
+
+   })
+
+ };
+ hideOrShow=()=>{
+   this.setState({condition:!this.state.condition})
+ };
+
+ handleSearch=(event)=>{
+   console.log(event.currentTarget.value)
+   let foodToSearch = event.currentTarget.value
+   let clonedFoods = this.state.foods.filter((food)=>{
+     return food.name.toLowerCase().includes(foodToSearch.toLowerCase())
+   })
+   this.setState({
+     filteredFood:clonedFoods
+   })
+
+ };
+
+ handleAdding=(a, b, c)=>{
```

Comment:
this is good approach for iteration 5. however, instead of adding the elements in an array. Best apprach would be to create one object with all properties (name, amount, calories) and then insert the object into the todaysFood state. That way when you map over the todayObject array in L67, you know each element of the map will be each food object added to the array.